src/otg/dataset/study_index.py

```diff
@@ -477,3 +477,113 @@ def from_source(
                 )
             )
         )
+
+
+@dataclass
+class StudyIndexUKBiobank(StudyIndex):
+    """Study index dataset from UKBiobank.
+
+    The following information is extracted:
+
+    - studyId
+    - pubmedId
+    - publicationDate
+    - publicationJournal
+    - publicationTitle
+    - publicationFirstAuthor
+    - traitFromSource
+    - ancestry_discoverySamples
+    - ancestry_replicationSamples
+    - initialSampleSize
+    - nCases
+    - replicationSamples
+
+    Some fields are populated as constants, such as projectID, studyType, and initial sample size.
+    """
+
+    @classmethod
+    def from_source(
+        cls: type[StudyIndexUKBiobank],
+        ukbiobank_studies: DataFrame,
+    ) -> StudyIndexUKBiobank:
+        """This function ingests study level metadata from UKBiobank.
+
+        Args:
+            ukbiobank_studies (DataFrame): UKBiobank study manifest file loaded in spark session.
+
+        Returns:
+            StudyIndexUKBiobank: Annotated UKBiobank study table.
+        """
+        return cls(
+            _df=(
+                ukbiobank_studies.select(
+                    f.col("code").alias("studyId"),
+                    f.col("trait").alias("traitFromSource"),
+                    f.col("n_cases").cast("long").alias("nCases"),
+                    f.col("n_total").cast("string").alias("initialSampleSize"),
+                    f.col("in_path").alias("summarystatsLocation"),
```

Comment:
When I try to access the sumstats from the gs:// links generated by this part, I'm getting:

```
$ gsutil ls gs://genetics-portal-raw/uk_biobank_sumstats/harmonised_split_190318/100001_raw.neale2.gwas.imputed_v3.both_sexes.GRCh38.tsv.split*.gz
CommandException: One or more URLs matched no objects.
```

(This URL is from the first row in the PR description)

Reply:
So in this case, I wonder if they have an updated manifest file somewhere?